src/WikiService.Data/RubricRepository.cs

```diff
@@ -112,5 +89,34 @@ public async Task<bool> DoesRubricNameExistAsync(Guid? rubricParentId, string na
     {
       return await _provider.Rubrics.AnyAsync(p => p.ParentId == rubricParentId && p.Name == nameRubric);
     }
+
+    private async Task<List<DbRubric>> FindRubricChild(IQueryable<DbRubric> dbRubrics)
+    {
+      foreach (DbRubric topRubric in dbRubrics.ToList())//todo - to rewrite the next foreach with Any()
+      {
+        foreach (DbRubric rubric in _provider.Rubrics.AsEnumerable())
+        {
+          if (rubric.ParentId == topRubric.Id)
+          {
+            topRubric.HasChild = true;
+            break;
+          }
+        }
+
+        if (!topRubric.HasChild)
+        {
+          foreach (DbArticle article in _provider.Articles.AsEnumerable())
+          {
+            if (article.RubricId == topRubric.Id)
+            {
+              topRubric.HasChild = true;
+              break;
+            }
+          }
+        }
+      }
+
+      return await dbRubrics.ToListAsync();
+    }
```

Comment:
you can add List<DbRubric> dbRubricsResult = await dbRubrics.ToListAsync() to 95'th line and use this variable in the first foreach and in return
this reduces the amount of db requests - instead of doing dbRubrics.ToList() in foreach and await dbRubrics.ToListAsync() in return you do it once in the method's start